opendatakit/briefcase
# Fix test overwriting user preferences

Closes #275 

This PR changes the Export tab to rely on the `FormCache` instead of reading the filesystem in search of form files.

This enables us to:
- ignore any concern regarding the Briefcase Storage directory, which is managed in an appScoped user preferences node.
- easily create an in-memory `FormCacheable` test double for our tests

#### What has been done to verify that this works as intended?
Starting with a well configured and running Briefcase app on my machine:
- Run `./gradlew build` without skipping the tests
- Run `java -jar build/libs/ODK[...].jar`
- Confirm that the Briefcase Storage dir is intact
- Confirm that the rest of functionality on the Export tab still works.

Also, automated test.

#### Why is this the best possible solution? Were any other approaches considered?
N/A

#### Are there any risks to merging this code? If so, what are they?
N/A
